cpp/src/IceGrid/ReapThread.cpp

```diff
@@ -137,24 +135,19 @@ ReapThread::join()
 }
 
 void
-ReapThread::add(
-    const shared_ptr<Reapable>& reapable,
-    chrono::seconds timeout,
-    const shared_ptr<Ice::Connection>& connection)
+ReapThread::add(const shared_ptr<Reapable>& reapable, chrono::seconds timeout, const Ice::ConnectionPtr& connection)
 {
     lock_guard lock(_mutex);
     if (_terminated)
     {
         return;
     }
 
-    //
-    // NOTE: registering a reapable with a null timeout is allowed. The reapable is reaped
-    // only when the reaper thread is shutdown.
-    //
+    // NOTE: registering a reapable with a 0s timeout is allowed. The reapable is reaped only when the reaper thread is
+    // shutdown or the connection is closed.
 
     //
-    // 10 seconds is the minimum permissable timeout.
+    // 10 seconds is the minimum permissable timeout (for non-zero timeouts).
     //
     if (timeout > 0s && timeout < 10s)
```

Comment:
I find odd that we have this minimum 10s buried here, and silently adjusted. We should move this out to the place we configure the timeouts. (for a follow-up PR)